faq/conda.md

```diff
@@ -0,0 +1,133 @@
+
+After installation, you can activate the base environment using
+
+```sh
+source $HOME/miniforge3/bin/activate
+```
+
+In order to initialize the environment whenever starting the bash shell, run
+
+```sh
+conda init --all
+```
+
+Now you can use `conda install` or `pip install` command to install conda or pip packages.
+
+## What is the difference between conda and pip?
+
+The largest difference is that pip can only install Python packages,
+but [conda](https://docs.conda.io/projects/conda/) can install any packages including C, C++, and Fortran packages without Python involved.
+We suggest reading the [Anaconda blog](https://www.anaconda.com/blog/understanding-conda-and-pip) for more information.
+
+## How can I check what packages have been installed?
+
+Run `conda list` and `pip list`, which show all conda/pip packages that have been installed in the current environment.
+
+## How to create a new conda/pip environment?
+
+To create a conda/pip environment named `my_env`, run
+
+```sh
+conda create -n my_env python=3.11
+```
+
+Now you can activate the environment using
+
+```sh
+conda activate my_env
+```
+
+If you want to activate the environment in a shell script, using `shell` instead:
+
+```sh
+source activate my_env
+```
+
+## Why do I need multiple environments instead of one?
+
+Due to multiple reasons, dependencies of packages often conflict with each other.
+For example, Python 3.11 and 3.12 cannot be installed into the same environment;
+CUDA 11 and CUDA 12 packages cannot be installed into the same environment;
+Open MPI and MPICH cannot be installed into the same environment;
+packages built against zlib 1.2 and 1.3 cannot be installed into the same environment.
+While [conda-forge](https://conda-forge.org) has tried its best to reduce package conflicts, it may still happen when installing multiple packages.
+Installing different packages into different environments will reduce the risk to meet the conflict packages.
+
+## How to use conda to install CUDA packages without CUDA installed?
+
+Usually, `conda` detects the CUDA driver on the system to install the [compatible](https://docs.nvidia.com/deploy/cuda-compatibility/index.html) CUDA packages.
+However, sometimes, you may want to install CUDA packages into a machine that doesn't install NVIDIA drivers (for example, the login node of a HPC cluster).
+In this case, you must manually set the environment variable `CONDA_OVERRIDE_CUDA` to tell conda which packages are compatible.
+
+For example, you want to use the installed CUDA packages on another machine where NVIDIA drivers have been installed. Run `nvidia-smi` on that machine, 
+```
++---------------------------------------------------------------------------------------+
+| NVIDIA-SMI 545.29.06              Driver Version: 545.29.06    CUDA Version: 12.3     |
+|-----------------------------------------+----------------------+----------------------+
+```
+the output shows the CUDA version is `12.3`.

```

Comment:
nvidia-smi shows the highest version of CUDA supported by the driver.

```suggestion
the output shows the highest supported CUDA version is `12.3`.
```

Reply:
I don't think so... It just shows the CUDA driver version (not the NVIDIA driver). But you can use CUDA toolkit 12.3 when the CUDA driver 12.0 is installed; and you can use CUDA toolkit 11.8 when the CUDA driver 11.2 is installed.

Reply:
Perhaps this is another necessary FAQ for CUDA things 🤣

Reply:
> I don't think so... It just shows the CUDA driver version (not the NVIDIA driver). 

I'm not sure if I got you right... [This post
](https://stackoverflow.com/questions/53422407/different-cuda-versions-shown-by-nvcc-and-nvidia-smi) said that.

> But you can use CUDA toolkit 12.3 when the CUDA driver 12.0 is installed; and you can use CUDA toolkit 11.8 when the CUDA driver 11.2 is installed.

That's right, and even the CUDA Toolkit is missing (only driver present), conda still installs packages with CUDA support correctly.

Anyway it's all up to you 🤣

Reply:
Oh, I think these posts are pretty old. NVIDIA introduced compatibility in CUDA 11.

Reply:
See https://docs.nvidia.com/deploy/cuda-compatibility/index.html#deployment-considerations